tools/build_api.py

```diff
@@ -527,8 +527,12 @@ def build_project(src_paths, build_path, target, toolchain_name,
         resources = scan_resources(src_paths, toolchain, inc_dirs=inc_dirs)
 
         # Generate PSA XML box code from manifests
-        psa_files_dir = process_manifest_files(resources.psa_manifests, build_path)
-        resources.add(toolchain.scan_resources(psa_files_dir))
+        detected_psa_sources = [f for f in resources.c_sources if re.match('.*\/spm\/src\/spm_.+\.c', f)]
+        detected_psa_sources += [f for f in resources.objects if re.match('.*\/spm\/src\/spm_.+\.o', f)]
```

Comment:
it will be uneficient to do perform this kind of check for larger code bases.
1. compile regex once and reuse in a loop
2. i would not depepnd on path separators stile as we also need to support Windows builds
3. use `any()` in conjunction with generator expression to stop iteration on first hit (as explained [here](https://stackoverflow.com/a/16505590/950970)) - this will prevent scanning all the files on first hit.
4. do we need to iterate over both c_sources and objects? will it be sufficient to iterate over object files only?
To summarize - i would write it in a following way:
```python
import itertools

pattern = re.compile('spm_.+\.[oc]')
detected_psa_sources = any(pattern.search(f) for f in itertools.chain(resources.objects, resources.c_sources))
if detected_psa_sources:
    ...
```